src/mesh/coordinates.cxx

```diff
@@ -249,57 +268,165 @@ Coordinates::Coordinates(Mesh *mesh, const CELL_LOC loc, const Coordinates* coor
       G3_23(mesh), G1(mesh), G2(mesh), G3(mesh), ShiftTorsion(mesh),
       IntShiftTorsion(mesh), localmesh(mesh), location(loc) {
 
-  dx = interpolateAndNeumann(coords_in->dx, location);
-  dy = interpolateAndNeumann(coords_in->dy, location);
+  std::string suffix = "";
+  switch (location) {
+  case CELL_XLOW: {
+      suffix = "_xlow";
+      break;
+    }
+  case CELL_YLOW: {
+      suffix = "_ylow";
+      break;
+    }
+  case CELL_ZLOW: {
+      // geometrical quantities are Field2D, so CELL_ZLOW version is the same
+      // as CELL_CENTRE
+      suffix = "";
```

Comment:
Is it worth implementing this now? @d7919 @bshanahan opinions?

Reply:
One reason not to implement now is that ideally we would treat 2d and 3d gridfiles differently here (and we don't currently have a flag to switch between the two). We want to read the metrics from the gridfile whenever possible (see #1607), and if we make `suffix` `_zlow` here, then a 2d grid file would have to store an extra set of fields (that are just duplicates of the `CELL_CENTRE` fields) to avoid calling `interpolateAndNeumann` (or `interpolateAndExtrapolate` if #1607 goes in) which would overwrite the boundary cells. The point being that we want `suffix = ""` for 2d grid files (so we're not just wasting disk space) but `suffix = "_zlow"` for 3d grid files. [although this is all academic as I think nobody's using CELL_ZLOW at the moment]

We could add a method to `GridDataSource` like `GridDataSource::isAxisymmetric` in anticipation of 3d metrics. For now it would just `return true`?

Reply:
Could look something like this: https://github.com/boutproject/BOUT-dev/compare/read-staggered-grids...read-staggered-grids-isaxisymmetric-switch

Reply:
Ok, I'm happy to leave it as-is for now, and we'll revisit when the 3D metrics stuff goes in.